src/lpython/semantics/python_comptime_eval.h

```diff
@@ -650,6 +653,43 @@ struct PythonIntrinsicProcedures {
 
     }
 
+    static ASR::expr_t *eval__lpython_str_capitalize(Allocator &al, const Location &loc, Vec<ASR::expr_t *> &args) {
+        LFORTRAN_ASSERT(ASRUtils::all_args_evaluated(args));
+        if (args.size() != 1) {
+            throw SemanticError("_lpython_str_capitalize() takes exactly one arguments (" +
+                std::to_string(args.size()) + " given)", loc);
+        }
+        ASR::expr_t *arg1 = args[0];
+        ASR::ttype_t *arg1_type = ASRUtils::expr_type(arg1);
+        if (ASRUtils::is_character(*arg1_type)) {
+            std::string val = ASR::down_cast<ASR::StringConstant_t>(arg1)->m_s;
+            ASR::ttype_t *type = ASRUtils::TYPE(ASR::make_Character_t(al, loc,
+                                        1, 1, nullptr, nullptr, 0));
+            ASR::ttype_t *res_type = ASRUtils::TYPE(ASR::make_StringConstant_t(al, loc, s2c(al, ""), type));
+            return ASR::down_cast<ASR::expr_t>(ASR::make_StringConstant_t(al, loc, s2c(al, val),  res_type));
```

Comment:
Shouldn't the code here implement the capitalization at compile time? It seems it just repackages the first argument as `StringConstant`, but does not capitalize.

Reply:
No, just keep adding commits until it is completely reviewed and ready. Only as the very last thing before merge you can rebase /  polish the history.